packages/kbn-scout/src/playwright/fixtures/worker/synthtrace.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type {
+  ApmSynthtraceEsClient,
+  OtelSynthtraceEsClient,
+  InfraSynthtraceEsClient,
+} from '@kbn/apm-synthtrace';
+import { Readable } from 'stream';
+import type { ApmFields, InfraDocument, OtelDocument } from '@kbn/apm-synthtrace-client';
+import Url from 'url';
+import {
+  getApmSynthtraceEsClient,
+  getInfraSynthtraceEsClient,
+  getOtelSynthtraceEsClient,
+} from '../../../common/services/synthtrace';
+import { coreWorkerFixtures } from './core_fixtures';
+import type { SynthtraceEvents } from '../../global_hooks/synthtrace_ingestion';
+
+export interface SynthtraceFixture {
+  apmSynthtraceEsClient: {
+    index: (events: SynthtraceEvents<ApmFields>) => Promise<void>;
+    clean: ApmSynthtraceEsClient['clean'];
+  };
+  infraSynthtraceEsClient: {
+    index: (events: SynthtraceEvents<InfraDocument>) => Promise<void>;
+    clean: InfraSynthtraceEsClient['clean'];
+  };
+  otelSynthtraceEsClient: {
+    index: (events: SynthtraceEvents<OtelDocument>) => Promise<void>;
+    clean: OtelSynthtraceEsClient['clean'];
+  };
+}
```

Comment:
Suggestion: could we try to avoid some duplication here? 

```suggestion
interface SynthtraceFixtureEsClient<TFields extends Fields> {
  index: (events: SynthtraceEvents<TFields>) => Promise<void>;
  clean: SynthtraceEsClient<TFields>['clean'];
}

export interface SynthtraceFixture {
  apmSynthtraceEsClient: SynthtraceFixtureEsClient<ApmFields>;
  infraSynthtraceEsClient: SynthtraceFixtureEsClient<InfraDocument>;
  otelSynthtraceEsClient: SynthtraceFixtureEsClient<OtelDocument>;
}
```

Reply:
done